crates/core/tedge_agent/src/agent.rs

```diff
@@ -97,10 +103,19 @@ impl AgentConfig {
         let http_bind_address = tedge_config.http.bind.address;
         let http_port = tedge_config.http.bind.port;
 
-        let http_config = HttpConfig::default()
-            .with_data_dir(data_dir.clone())
-            .with_port(http_port)
-            .with_ip_address(http_bind_address);
+        let rustls_config = if let Some((cert, key)) = load_certificate_and_key(&tedge_config)? {
+            let ca_path = tedge_config.http.ca_path.or_none();
+            let root_cert_store = ca_path.map(|s| axum_tls::read_trust_store(s)).transpose()?;
+            Some(axum_tls::ssl_config(cert, key, root_cert_store)?)
+        } else {
+            None
+        };
```

Comment:
suggestion: I feel like that's not quite the right place to put loading and parsing of certificates and cryptographic keys.

IMO the main reason why we e.g. don't use `TedgeConfig` directly inside the actors, is because then to create actors we'd have to prepare the entire `TedgeConfig` with many config settings that do not affect the actor we're trying to start. It would make the actor harder to use/test in isolation, outside of the context of its primary user (e.g. File Transfer Service HTTP server inside tedge-agent).

I think loading and parsing certificates and private keys in this `TedgeConfig -> ActorConfig` transformation also makes the actor harder to use, because then to create `ActorConfig` directly, you have to manually produce `rustls::ServerConfig`. It wouldn't be that much code, thanks to the helper functions you made, but still, I'd argue that it's a bit wrong level of abstraction to do that here.

Instead, we should pass paths themselves to `FileTransferServerBuilder` itself.

Reply:
Agree. A function creating a config object should be as light-weight as possible.